doc/services/shell/index.rst

```diff
@@ -690,9 +690,11 @@ in the FreeBSD project. This feature is activated by:
 :kconfig:option:`CONFIG_POSIX_C_LIB_EXT` set to ``y`` and :kconfig:option:`CONFIG_GETOPT_LONG`
 set to ``y``.
 
-This feature can be used in thread safe as well as non thread safe manner.
-The former is full compatible with regular getopt usage while the latter
-a bit differs.
+This feature can be used in thread safe as well as non thread safe
+manner.  The former is full compatible with regular getopt usage while
+the latter differs a bit. To gain access to the thread-safe APIs, add
+`#define __need_getopt_newlib` before `#include <getopt.h>` in your
```

Comment:
I would guess that this only applies to newlib and picolibc, since other C libraries likely would not key off of newlib-specific defines.

Reply:
We're using the newlib extensions to the BSD/Glibc API specification here, which defines what appears in `<getopt.h>`. Much like when we extend POSIX with BSD/Glibc symbols in POSIX headers, the newlib extensions only appear when the application defines `__need_getopt_newlib` before including `<getopt.h>`. For Zephyr to be compatible with these newlib extensions, it should follow the same rules, otherwise Zephyr applications might fail to work with newlib or picolibc.